server/src/__tests__/utils/mikro-integration-test-environment.ts

```diff
@@ -0,0 +1,54 @@
+import { mocks } from '../../test-utils'
+import { db, app, http } from '../../setup'
+import { ClubEntity } from '../../shared/infra/db/entities/club.entity'
+import { EventEntity } from '../../shared/infra/db/entities/event.entity'
+import { TagEntity } from '../../shared/infra/db/entities/tags/tag.entity'
+
+export interface Variables {
+  webServer: http.WebServer
+}
+
+// This is similar to Jest environments.
+// We don't use Jest environments here because they modify the global object in a type-unsafe way.
+class MikroIntegrationTestEnvironment {
+  protected mikroDB!: db.MikroDB
+
+  protected ids!: Array<string>
```

Comment:
Curious why these are protected if the class is never inherited from.

Reply:
The idea is if someone were to inherit from the class, they might want to use these fields in whatever behaviour they modify